packages/eslint-plugin/docs/rules/no-invalid-void-type.md

```diff
@@ -97,6 +99,23 @@ type AllowedVoid = Ex.Mx.Tx<void>;
 type AllowedVoidUnion = void | Ex.Mx.Tx<void>;
 ```
 
+#### `allowAsThisParameter`
+
+This option allows specifying a `this` parameter of a function to be `void` when set to `true`.
+TypeScript [recommends using this pattern](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters-in-callbacks) to explicitly label function types that do not use a this argument.
```

Comment:
So [I asked the team for clarification](https://github.com/microsoft/TypeScript/issues/40704) and got confirmation that using a `this` **parameter** without the `this` **type** doesn't have significant performance impact. Would it be fine if I just say "typescript *uses* this pattern" instead of saying *recommends*?

Reply:
yeah, I want to be careful with the docs. I've learned that some people treat stuff that's written in popular projects like this as a bible for what to do / what not to do.

When we were small, I used to rampantly push my own style and agenda, but as this has grown, I've tried to be less and less prescriptive. We are used by roughly ~44% of TS users.

Let's split the difference. Something like:

```
This pattern can be useful to explicitly label function types that do not use a `this` argument. [See the TypeScript docs for more information](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters-in-callbacks).
```

Reply:
I'm good with that.

```suggestion
This pattern can be useful to explicitly label function types that do not use a `this` argument. [See the TypeScript docs for more information](https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters-in-callbacks).
```

Reply:
ok yeah I see it now, "recommends" means there will be people who blindly apply it to all their methods, got it.